internal_worker_base.go

```diff
@@ -141,33 +150,42 @@ func (bw *baseWorker) Stop() {
 	// TODO: The poll is longer than the 10 seconds, we probably need some way to hard terminate the
 	// poll routines as well.
 
-	if success := awaitWaitGroup(&bw.shutdownWG, 10*time.Second); !success {
+	if success := awaitWaitGroup(&bw.shutdownWG, 2*time.Second); !success {
```

Comment:
Note to reviewers: I changed this to 2 seconds. No reason for Stop to take very long. And after the bug fix that is part of this PR, Stop usually happens very quickly < 1 sec.